### PR TITLE
Osx fix

### PR DESCRIPTION
**Pull request by egaburov**
Build C++ examples on OSX:

    $ clang++ -std=c++11 saxpy.cpp -I./  && ./a.out       
    OK
    $ clang++ -std=c++11 sort.cpp -I./  && ./a.out     
    OK
    $ clang++ -std=c++11 test_shared.cpp  -I./  && ./a.out     
    [works with a lot of output]

But getting error form CUDA examples:

    $ nvcc -std=c++11 saxpy.cu  -I./  && ./a.out                                      
    ./agency/detail/tuple_impl.hpp(741): error: namespace "std" has no member "get"
    
    ./agency/detail/tuple_impl.hpp(747): error: namespace "std" has no member "get"
    
    ./agency/detail/tuple_impl.hpp(753): error: namespace "std" has no member "get"
    
    3 errors detected in the compilation of "/var/folders/3q/bvdwl5dj72g53sldd774gdq80000gn/T//tmpxft_00009ad0_00000000-9_saxpy.cpp1.ii".

Any idea why it is so? It appears to be of the similar nature as Intel C++ compiler produced on fixed tuple (bottom of https://github.com/jaredhoberock/tuple/issues/26)

### CODE REVIEW

**Comment by egaburov** [comment]
Here is a simple reproducer that triggers this error:

    $ cat test.cpp
    #include <utility>
    #define __TUPLE_ANNOTATION
    #define __TUPLE_NAMESPACE foo
    namespace __TUPLE_NAMESPACE                                                                                    
    { 
      template<class... Types>                                                                                     
        class tuple
        {   
      public:
        template<size_t i, class... UTypes>
          friend __TUPLE_ANNOTATION                                                                                
              typename std::tuple_element<i, __TUPLE_NAMESPACE::tuple<UTypes...>>::type
              std::get(__TUPLE_NAMESPACE::tuple<UTypes...>& t);
    
        };
    };
    int main() {}
    $ g++ test.cpp -std=c++11
    [no error]
    $ nvcc -x cu test.cpp -std=c++11
    test.cpp(13): error: namespace "std" has no member "get"
    
    1 error detected in the compilation of "/var/folders/3q/bvdwl5dj72g53sldd774gdq80000gn/T//tmpxft_00009ced_00000000-9_test.cpp1.ii".

On Linux, nvcc compiles this reproducer w/o errors.

**Comment by egaburov** [comment]
Work-aroudn for the test problem above requires using header files form gcc. On my mac, using g++-mp-4.9 form macports:
  
    $ nvcc -x cu test.cpp -std=c++11 -I/opt/local/include/gcc49/c++ -I/opt/local/include/gcc49/c++/x86_64-apple-darwin14
   [no error]

Can compile saxpy.cu in `agency` example with:
  
    $ nvcc -c -std=c++11 saxpy.cu  -I./ -I/opt/local/include/gcc49/c++ -I/opt/local/include/gcc49/c++/x86_64-apple-darwin14 
    [no error]

but linking produces tons of errors. Still investigating.

**Comment by jaredhoberock** [comment]
Does this still need to be merged? It doesn't look like there are any differences between this branch and master.

File: agency/coordinate.hpp [hunk 1]
```diff
@@ -796,8 +796,8 @@ namespace std
 {
 
 
-template<size_t I, class Tuple>
-struct tuple_element;
+//template<size_t I, class Tuple>
+//struct tuple_element;
```
Comment (jaredhoberock) [inline agency/coordinate.hpp]: If this declaration is causing problems, it should be completely eliminated, not just commented out.

**Comment by egaburov** [comment]
done!

**Pull request merged**